lib/auth.js

```diff
@@ -52,6 +52,26 @@ var auth = {
     } else {
       return cb(null, false);
     }
+  },
+  /**
+   * New endpoint required as autentication ($fh.auth) requests not handled
+   * correctly in local development/Studio Preview -
+   * the js sdk dertermines envrironment mode from url string.
+   * New endpoint determines environment mode from env var,
+   * sends request (req) and calls user provided function (cb).
+   * @param  {object} req
+   * @param  {object} res
+   * @param  {object} cb
+   */
+  getEnvironmentMode: function (req, res, cb) {
```

Comment:
not sure about the name of the function, seems a bit confusing. Perhaps just call it `performAuth`?